tests/test_taskgroups.py

```diff
@@ -542,6 +542,26 @@ async def cancel_when_ready():
     assert not sleep_completed
 
 
+async def test_shielding_mutate():
+    done = False
+
+    async def task(task_status):
+        nonlocal done
+        with CancelScope() as scope:
+            await sleep(.1)
+            scope.shield = True
+            task_status.started()
+            await sleep(.1)
+            scope.shield = False
+            done = True
```

Comment:
The point is to check if `scope.shield=False` cancels the task. If the task exits right after setting `shield` to `False`, nothing is tested.

Reply:
that's right-- I've reverted to your suggested code, thanks for your patience